apixu-weatherbot1.0.tcl

```diff
@@ -0,0 +1,498 @@
+            puthelp "PRIVMSG $chan :$errormsg"
+            return
+        }
+
+        puthelp "PRIVMSG $chan :$current_weather"
+    }
+
+    proc forecast {nick uhost hand chan text} {
+        set text [string trim $text]
+
+        if {$text eq "--help" || $text eq "-h"} {
+            _get_help $nick
+            return
+        }
+
+
+        # Checks to see if a user has a location set
+        set userinfo [_get_userinfo $nick $hand]
+        if {$text eq "" && [dict exists $userinfo error]} {
+            if {$::weather::private} {
+                puthelp "NOTICE $nick :Did you want the weather for a specific location?\
+                         Or please PM me with the \".set\" command to set a default location."
+                return
+            }
+
+            puthelp "PRIVMSG $chan :No location set or use \002'.wzf <location>'\002"
+            return
+        } elseif {[dict exists $userinfo error]} {
+            set userinfo [dict create location $text units $::weather::units]
+        } elseif {$text ne ""} {
+            set userinfo [dict create location $text units [dict get $userinfo units]]
+        }
+
+        set location [dict get $userinfo location]
+        if {[catch {set json_data [_get_json $location "forecast"]} errormsg]} {
+            puthelp "PRIVMSG $chan :$errormsg"
+            return
+        }
+
+        if {[catch {set forecastdays [_json_parse $json_data $userinfo "forecast"]} errormsg]} {
+            puthelp "PRIVMSG $chan :$errormsg"
+            return
+        }
+
+        puthelp "PRIVMSG $chan :$forecastdays"
+    }
+
+    proc location {nick uhost hand chan text} {
+        set $text [string trim $text]
+
+        if {$text eq "--help" || $text eq "-h"} {
+            _get_help $nick
+            return
+        } elseif {$::weather::private} {
+            # If $private is set to 1, send a PM
+            putlog "::weather::private set to private. Use PM instead."
+            puthelp "PRIVMSG $nick :Please private message me to set your location.\
+                     i.e. \002'.set 1 <location>'\002 to set your location and use imperial\
+                     units."
+            return
+        }
+
+        if {[catch {set location_info [_set_location $nick $uhost $hand $text]} errormsg]} {
+            puthelp "PRIVMSG $chan :$errormsg"
+            return
+        }
+
+        puthelp "PRIVMSG $chan :Default weather location for \002$nick\002 set to\
+                 \002[dict get $location_info location]\002 and units set to\
+                 \002[dict get $location_info units]\002\."
+    }
+
+    ###################################
+    # Private message command functions
+    ###################################
+    proc pm_current {nick uhost hand text} {
+        set text [string trim $text]
+
+        if {$text eq "--help" || $text eq "-h"} {
+            _get_help $nick
+        }
+    }
+
+    proc pm_forecast {nick uhost hand text} {
```

Comment:
does this do anything? I don't see it forecasting but maybe I am missing something

Reply:
pm_current and pm_forecast was set there in case someone wanted to call for help in PM, that way they don't have to go back to the channel to call it if they are in the PM already. Maybe later be used to call weather in PM , too.